File: book_review_web/backend/src/main/java/com/bookreview/service/BookService.java
package com.bookreview.service;

import com.bookreview.dto.BookDTO;
import com.bookreview.dto.PageResponseDTO;

public interface BookService {
    PageResponseDTO<BookDTO> getAllBooks(int page, int size);
    BookDTO createBook(BookDTO bookDTO);
    BookDTO updateBook(Long id, BookDTO bookDTO);
    void deleteBook(Long id);
}

File: book_review_web/frontend/src/services/bookService.js
import api from './api';

export const bookService = {
  getBooks: async (page = 0, size = 10) => {
    const response = await api.get('/books', { params: { page, size } });
    return response.data;
  },
  
  createBook: async (data) => {
    const response = await api.post('/books', data);
    return response.data;
  },
  
  updateBook: async (id, data) => {
    const response = await api.put(`/books/${id}`, data);
    return response.data;
  },
  
  deleteBook: async (id) => {
    const response = await api.delete(`/books/${id}`);
    return response.data;
  }
};
